wasi/wasi.go

```diff
@@ -38,7 +40,7 @@ type API interface {
 	//   * ArgsSizesGet result argv_buf_size bytes are written to this offset
 	//
 	// For example, if ArgsSizesGet wrote argc=2 and argvBufSize=5 for arguments: "a" and "bc"
-	//   and ArgsGet results argv=7 and argvBuf=1, we expect `ctx.Memory.Buffer` to contain:
+	//   and ArgsGet parameters are argv=7 and argvBuf=1, we expect `ctx.Memory.Buffer` to contain:
```

Comment:
spec is so crap that sometimes out params (offsets to write data) are classified as params and other times results. this is ok how you've interpreted it!